### DISCUSSION TITLE
[A/L] (2021/22) Funzioni - Esercizio 20

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/metodi-matematici-per-l-informatica#260] [Esercizi A-L]
![image](https://user-images.githubusercontent.com/69389461/152316345-b3310f9b-566e-4cbb-807e-05af5825ece7.png)

**Comment by LuginiAndrea** [comment]
Sinceramente non ho ben inteso cosa faccia $f$, ma possiamo comunque dire alcune cose.
Primo, $f$ non è sempre iniettiva. Infatti sappiamo che data una funzione $g: X \rightarrow Y$, suddetta funzione non può esserre iniettiva se Card( $Y$ ) < Card( $X$ ), il che ne deriva che per essere iniettiva $g$ deve avere Card( $X$ ) $\leq$ Card( $Y$ ). 
Sappiamo che Card( $X \times Y$ ) = Card( $X$ ) $\times$ Card( $Y$ ), e che Card( $P\left(X\right)$ ) = $2^{\textrm{Card}\left(X\right)}$ (ovviamente parliamo di insiemi finiti, altrimenti dovremmo considerare l'algebra dei numeri trascendentali).
Quindi la cardinalità di $P\left(X \times Y\right) = 2^{\textrm{Card}\left(X\right) \times \textrm{Card}\left(Y\right)}$.
La cardinalità di $P\left(X\right) \times P\left(Y\right) = 2^{\textrm{Card}\left(X\right)} \times 2^{\textrm{Card}\left(Y\right)} = 2^{\textrm{Card}\left(X\right) + \textrm{Card}\left(Y\right)}$
Basti considerare quindi quando Card( $X$ ) $\times$ Card( $Y$ ) $<$ Card( $X$ ) + Card( $Y$ ) per avere casi in cui $f$ non è iniettiva.

Con ragionamento analogo, sapendo che una funzione $f: X \rightarrow Y$ per essere suriettiva deve avere che Cardinalità $X \geq$ Cardinalità $Y$.
Al pari di prima, basta considerare quando Card( $X$ ) $\times$ Card( $Y$ ) $>$ Card( $X$ ) + Card( $Y$ ) per avere casi in cui $f$ non è suriettiva.
Un esempio è $X = 10, 20, 30$ e $Y = 50, 60$, senza dilungarmi ad esporlo per lungo
